File: app/src/main/java/com/kota/Telnet/Model/TelnetData.java
package com.kota.Telnet.Model;

public class TelnetData {
    private static int _count = 0;
    public byte data = 0;
    public byte textColor = 0;
    public byte backgroundColor = 0;
    public boolean blink = false;
    public boolean italic = false;

    public TelnetData() {
    }

    public TelnetData(TelnetData aData) {
        set(aData);
    }

    public void set(TelnetData aData) {
        this.data = aData.data;
        this.textColor = aData.textColor;
        this.backgroundColor = aData.backgroundColor;
        this.blink = aData.blink;
        this.italic = aData.italic;
    }

    public TelnetData clone() {
        return new TelnetData(this);
    }

    public void clear() {
        this.data = (byte) 0;
        this.textColor = (byte) 0;
        this.backgroundColor = (byte) 0;
        this.blink = false;
        this.italic = false;
    }

    public boolean isEmpty() {
        return this.data == 0;
    }
}

File: app/src/main/java/com/kota/Telnet/Model/TelnetData.kt
package com.kota.Telnet.Model

data class TelnetData(
    var data: Byte = 0,
    var textColor: Byte = 0,
    var backgroundColor: Byte = 0,
    var blink: Boolean = false,
    var italic: Boolean = false
) {
    
    companion object {
        private var count = 0
    }
    
    constructor(data: TelnetData) : this() {
        set(data)
    }
    
    fun set(data: TelnetData) {
        this.data = data.data
        this.textColor = data.textColor
        this.backgroundColor = data.backgroundColor
        this.blink = data.blink
        this.italic = data.italic
    }
    
    public override fun clone(): TelnetData {
        return TelnetData(this)
    }
    
    fun clear() {
        data = 0
        textColor = 0
        backgroundColor = 0
        blink = false
        italic = false
    }
    
    fun isEmpty(): Boolean {
        return data == 0.toByte()
    }
}
